packages/signing-utils/src/signing-clients/local-signing-client.ts

```diff
@@ -0,0 +1,53 @@
+import path from 'path';
+import { exec } from 'child_process';
+import { promisify } from 'util';
+import { debug } from '../utils';
+import type { SigningClient, SigningClientOptions } from '.';
+
+const execAsync = promisify(exec);
+
+export class LocalSigningClient implements SigningClient {
+  constructor(private options: SigningClientOptions) {}
+
+  private async init() {
+    const remoteScript = `${this.options.rootDir}/garasign.sh`;
+    await execAsync(`mkdir -p ${this.options.rootDir}`);
+    await this.copyFile(this.options.signingScript, remoteScript);
+    await execAsync(`chmod +x ${remoteScript}`);
+  }
+
+  private async copyFile(from: string, to: string): Promise<void> {
+    await execAsync(`cp ${from} ${to}`);
+  }
+
+  async sign(file: string): Promise<void> {
+    debug('Signing file', file);
+
+    const remotePath = path.join(this.options.rootDir, path.basename(file));
+
+    try {
+      await this.init();
+
+      await this.copyFile(file, remotePath);
+      debug(`LocalSigningClient: Copied file ${file} to ${remotePath}`);
+
+      await execAsync(
+        `cd ${this.options.rootDir} && ./garasign.sh ${path.basename(file)}`,
+        {
+          env: {
+            method: this.options.signingMethod,
+          },
```

Comment:
I just figured out that we need to pass the garasign and artifactory env vars here. i added [685329f](https://github.com/mongodb-js/compass/pull/5279/commits/685329feba42bd608b0868097ff12990572bb43b) in original pr

Reply:
I believe I address this comment, the generic helper will gather all environment variables and I assign them here. Let me know if I missed something.